addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/ExponentialEntityExpansionScanRule.java

```diff
@@ -0,0 +1,168 @@
+                    + " <!ENTITY lol4 \"&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;&lol3;\">\n"
+                    + " <!ENTITY lol5 \"&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;&lol4;\">\n"
+                    + " <!ENTITY lol6 \"&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;&lol5;\">\n"
+                    + " <!ENTITY lol7 \"&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;&lol6;\">\n"
+                    + " <!ENTITY lol8 \"&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;&lol7;\">\n"
+                    + " <!ENTITY lol9 \"&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;&lol8;\">\n"
+                    + "]>\n"
+                    + "<lolz>&lol9;</lolz>";
+    private static final String YAML_PAYLOAD =
+            "a: &a [\"lol\",\"lol\",\"lol\",\"lol\",\"lol\",\"lol\",\"lol\",\"lol\",\"lol\"]\n"
+                    + "b: &b [*a,*a,*a,*a,*a,*a,*a,*a,*a]\n"
+                    + "c: &c [*b,*b,*b,*b,*b,*b,*b,*b,*b]\n"
+                    + "d: &d [*c,*c,*c,*c,*c,*c,*c,*c,*c]\n"
+                    + "e: &e [*d,*d,*d,*d,*d,*d,*d,*d,*d]\n"
+                    + "f: &f [*e,*e,*e,*e,*e,*e,*e,*e,*e]\n"
+                    + "g: &g [*f,*f,*f,*f,*f,*f,*f,*f,*f]\n"
+                    + "h: &h [*g,*g,*g,*g,*g,*g,*g,*g,*g]\n"
+                    + "i: &i [*h,*h,*h,*h,*h,*h,*h,*h,*h]";
+
+    @Override
+    public int getId() {
+        return 40044;
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString(PREFIX + "name");
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString(PREFIX + "desc");
+    }
+
+    @Override
+    public int getCategory() {
+        return Category.MISC;
+    }
+
+    @Override
+    public String getSolution() {
+        return Constant.messages.getString(PREFIX + "soln");
+    }
+
+    @Override
+    public String getReference() {
+        return Constant.messages.getString(PREFIX + "refs");
+    }
+
+    @Override
+    public int getRisk() {
+        return Alert.RISK_MEDIUM;
+    }
+
+    @Override
+    public Map<String, String> getAlertTags() {
+        return CommonAlertTag.toMap(
+                CommonAlertTag.OWASP_2021_A04_INSECURE_DESIGN,
+                CommonAlertTag.WSTG_V42_BUSL_09_UPLOAD_MALICIOUS_FILES);
+    }
+
+    @Override
+    public int getCweId() {
+        return 776;
+    }
+
+    @Override
+    public int getWascId() {
+        return 44;
+    }
+
+    @Override
+    public List<Alert> getExampleAlerts() {
+        return Collections.singletonList(newAlert().setConfidence(Alert.CONFIDENCE_LOW).build());
+    }
+
+    @Override
+    public void scan() {
+        HttpMessage msg = getBaseMsg();
+        String contentType = msg.getRequestHeader().getHeader(HttpHeader.CONTENT_TYPE);
+        if (contentType == null) {
+            return;
+        }
+        String attack;
+        if (contentType.contains("xml")) {
```

Comment:
Will case be problematic here? Is there an ignore case variant of contains?

Reply:
Same for yaml

Reply:
Good point, will update.

Reply:
It can use `HttpHeader#hasContentType` instead.